### PR TITLE
release-notes.txt: add 2020.04 release notes

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
This adds the notes for the 2020.04 release. ~~The statistics are still missing as one PR is waiting review and will be backported.~~

The list of changes is quite extensive, so please let me know if it should be shortened. A Kconfig migration section was added to inform of the current status of it and the changes made in that direction.

Please amend any missing  changes or removals!

### Testing procedure
Read the document

### Issues/PRs references
None

### CODE REVIEW

**Comment by kb2ma** [comment]
CoAP related items look accurate to me.

**Comment by leandrolanzieri** [comment]
Added the stats

File: release-notes.txt [hunk 1]
```diff
@@ -1,3 +1,775 @@
...
+    * add uquery improvements
+    * make separate tree handling function
+    - remove obsolete functions for addition of Uri-Query option
++ net/netdev: add netdev_trigger_event_isr() function
+- net/nhdp: remove module
++ net/sock:
+    + amend API to iterate over stack-internal buffer chunks
+    + amend with zero-copy receive functions
+    * async: add optional callback argument
+    * async_event: fix race-condition
```
Comment (miri64) [inline release-notes.txt:133]: After looking at it again, I actually would remove the last point (its a bug fix, not a feature) and then reverse the whole list. So add optional callback argument first ,then zero copy, then iteration

File: release-notes.txt [hunk 2]
```diff
@@ -1,3 +1,772 @@
...
+- Improvements on Xtimer module and testing
+- Initial addition of the new timer subsystem: Ztimer
+- Migration to Kconfig - phase I: add multiple GNRC and system modules
+- +10 new boards, +11 new drivers, +7 new packages
+
+
+About 555 pull requests, composed of 1514 commits, have been merged since the
+last release, and about 22 issues have been solved. 60 people contributed with
+code in 102 days. Approximately 3346 files have been touched with 358398 (+)
+insertions and 184720 deletions (-).
```
Comment (miri64) [inline release-notes.txt:38]: Very precise numbers while using words like "about" and "approximately" ;-)

**Comment by jia200x** [comment]
The `gnrc_netif` stuff seem accurate to me

File: release-notes.txt (hunk 2 above)
Comment (leandrolanzieri) [reply release-notes.txt]: Done

File: release-notes.txt [hunk 3]
```diff
@@ -1,3 +1,772 @@
...
+* make:
+    * disable implicit rules
+    * remove support for make <4
+    * native: turn on creation of debug symbols (CFLAGS += -g)
+* Makefile:
+    * arch: mips: Allow CFLAGS_DBG and CFLAGS_OPT to be overridden
+    * boards: move remaining uses of USEMODULE from Makefile.include to
+      Makefile.dep
+    * disable stdio_% modules before they are included
+    * even more useless exports removed
```
Comment (fjmolinas) [inline release-notes.txt:310]: Maybe remove this one
```suggestion
```

File: release-notes.txt [hunk 4]
```diff
@@ -1,3 +1,772 @@
...
+    * fix for `GNRC_NETIF_SINGLE`
+    * fixes for evtimer_msg and bench_runtime_coreapis
++ tests/bench_xtimer_load: initial commit
+* tests/buttons: fix build failure if BTN0_PIN is not declared
++ tests/candev: add initial version with native support
++ tests/driver_netdev_common: add compile-test for network drivers
++ tests/emcute: provide tests
++ tests/gnrc_ipv6_nib_dns: add test of RDNSS option handling
+* tests/gnrc_rpl_srh: use AsyncSniffer for sniffing
+* tests/periph_rtc: fix system locks in IS
```
Comment (fjmolinas) [inline release-notes.txt:338]:
```suggestion
* tests/periph_rtc: fix system locks in ISR
```

File: release-notes.txt [hunk 5]
```diff
@@ -1,3 +1,772 @@
...
+    * use generic exti_config
+* cpu/samd21: pwm: allow to use channels > 3
+* cpu/samd5x: disable RTC on init to prevent undefined RTC state
+* cpu/[saml21/1x]: enable buck voltage regulator when possible
+* cpu/stm32_common:
+    + add USB OTG FS/HS usbdev peripheral driver
+    * enable EXTI interrupt for rtt
+    * eth: fix address and multicast filtering
+* cpu/stm32f1: make RTC Y2038 safe
++ cpu/stm32wb: add support
```
Comment (fjmolinas) [inline release-notes.txt:252]:
```suggestion
+ cpu/stm32wb: add initial support
```

File: release-notes.txt [hunk 6]
```diff
@@ -1,3 +1,772 @@
...
+    * add CPU feature for stack smash protections
+    + add UART RX implementation on PIC32 devices
+    * make default idle/main stacksizes configurable on all archs
+    * move cpu level dependencies in dedicated Makefile.dep files
+* cpu/atmega_common: fix reboot issues
+* cpu/cc13x2: rename cpu to cc26x2_cc13x2
+* cpu/cc2538:
+    * fix spi_transfer_bytes()
+    + implement periph/pm
+    * timer: fix GPT enabling wait
```
Comment (fjmolinas) [inline release-notes.txt:202]: This one is a fix to `timer: handle power mode`, I would simply remove.
```suggestion
```

File: release-notes.txt [hunk 7]
```diff
@@ -1,3 +1,772 @@
...
+    + add 'Creating boards' section
+    * improvements on various board's documentation
+
+CPU
+---
+* cpu:
+    * add CPU feature for stack smash protections
+    + add UART RX implementation on PIC32 devices
+    * make default idle/main stacksizes configurable on all archs
+    * move cpu level dependencies in dedicated Makefile.dep files
```
Comment (dylad) [inline release-notes.txt:196]:
```suggestion
    * move cpu level dependencies in dedicated Makefile.dep files
    * unify UART hardware flow control use to a generic module
```

File: release-notes.txt [hunk 8]
```diff
@@ -1,3 +1,772 @@
...
++ net/dhcpv6: initial implementation of a client (with IA_PD support)
+* net/dns: fix DNS resolution in ping6
+* net/gcoap:
+    + add canonical uri query function names
+    + allow proxied client requests
+    * use sock_async and events
++ net/gnrc_dhcpv6_client_6lbr: initial import of a 6LBR DHCPv6 client
++ net/gnrc_ipv6_ext_frag: initial import of statistics module
+* net/gnrc_ipv6_nib:
+    * do not ignore PL for route
```
Comment (fjmolinas) [inline release-notes.txt:100]: This seems clearer

```suggestion
    * do not ignore prefix list for route resolution
```

@miri64 might want to check it though.

**Comment by leandrolanzieri** [comment]
@RIOT-OS/maintainers please take a last look

File: release-notes.txt [hunk 9]
```diff
@@ -1,3 +1,771 @@
...
+The 2020.04 release includes:
+
+- Add support for 6LoWPAN IPv6 extension header next header compression
+- Add support for DHCPv6 prefix delegation client
+- Add support for kw41zrf and at86rf215 IEEE 802.15.4 radios
+- Improvements on power management for ESP32
+- Improvements and extensions in support of fe310 and lpc2387
+- lwIP: provide support for asynchronous sock
+- Improvements on Xtimer module and testing
+- Initial addition of the new timer subsystem: Ztimer
```
Comment (cgundogan) [inline release-notes.txt:30]:
```suggestion
- Improvements on xtimer module and testing
- Initial addition of the new timer subsystem: ztimer
```

We should probably use the lowercase variants of `xtimer` and `ztimer` here to be consistent with the lower part of this document.

**Comment by leandrolanzieri** [comment]
@fjmolinas @cgundogan may I squash?

**Comment by leandrolanzieri** [comment]
Removed whitespace and fix spelling of the issue #13490 (also updated the issue's title :-)

**Comment by cgundogan** [comment]
@leandrolanzieri please go ahead and squash

**Comment by miri64** [comment]
Let's goooooo!

**Comment by leandrolanzieri** [comment]
Backport provided in #13983

**Pull request merged**